### PR TITLE
User can filter profiles by location.

### PR DESCRIPTION
**Pull request by johnhaup**
buildaband-issue-07-pr

### CODE REVIEW

File: band.sql [hunk 1]
```diff
@@ -3,10 +3,7 @@ DROP TABLE IF EXISTS users;
 CREATE TABLE users (
   id          serial,
   username    text,
+  pass        text,
   instruments text [],
-  styles      text [],
-  space       integer,
-  age         integer,
-  originals   integer,
```
Comment (victormejia) [inline band.sql]: no longer using these columns?

File: index.js [hunk 2]
```diff
@@ -39,6 +41,20 @@ app.get('/users', (req, res) => {
...
+  res.status(200).send(zip)
+})
+
+app.post('/sortByLoc', (req, res) => {
+  let curList = req.body
+  for (let i = 1; i < curList.length; i++) {
+    curList[i].proximity = zipcodes.distance(curList[0].zipcode, curList[i].zipcode)
+  }
+  res.status(200).send(curList)
+})
```
Comment (victormejia) [inline index.js]: I like this API, but let's continue with REST semantics here. Essentially, you are still getting `users` here, so you should reuse the `app.get('/users')` handler. However, the request you will be making will now be to a path like this:

```
/users?zipcode=90022
```

That's a query param, and you can easily retrieve its value: https://expressjs.com/en/4x/api.html#req.query

That said, your DB should be the source of truth, so it should return the users that are in proximity for the zip code that you want.

File: band.sql (hunk 1 above)
Comment (johnhaup) [inline band.sql]: If I finish the MVP features, I'm going to add these as a way to filter profiles.

Comment (victormejia) [inline band.sql]: 👍

File: index.js [hunk 3]
```diff
@@ -31,14 +33,26 @@ app.get('/users', (req, res) => {
   query
     .select('*').from('users')
     .then(table => {
-      res.status(200).send(table)
+      table.forEach(user => {
+        user.proximity = zipcodes.distance(req.query.zip, user.zipcode)
```
Comment (victormejia) [inline index.js]: does this work if a user doesn't pass in a zip code? In that case, all users should be returned

**Pull request merged**